src/server.rs

```diff
@@ -63,7 +78,12 @@ impl Server {
         let msg = recv_json(&mut stream, &mut buf).await?;
 
         match msg {
-            Some(ClientMessage::Hello(port)) => {
+            Some(ClientMessage::Hello((port, secret))) => {
+                if let Err(e) = self.authenticate(secret) {
+                    send_json(&mut stream, ServerMessage::ClientError(e.into())).await?;
+                    return Ok(());
```

Comment:
Could this be given a more descriptive name like `Unauthenticated` or `Invalid` rather than `ClientError`?